packages/xstate-react/test/useSpawn.test.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import { useActor } from '../src';
+import { render, cleanup, fireEvent } from '@testing-library/react';
+import { useSpawn } from '../src/useSpawn';
+import { fromReducer } from 'xstate/src/behavior';
+
+afterEach(cleanup);
+
+describe('useSpawn', () => {
+  it('should be able to spawn an actor from a behavior', () => {
+    const reducer = (state: number, event: { type: 'INC' }): number => {
+      if (event.type === 'INC') {
+        return state + 1;
+      }
+
+      return state;
+    };
+
+    const behavior = fromReducer(reducer, 0);
+
+    const Test = () => {
+      const actorRef = useSpawn(behavior);
+      const [count, send] = useActor(actorRef);
```

Comment:
Quite frankly - this doesn't sell me on those APIs. Why would I like to refactor the existing reducer to that? It would be great to list the benefits and possibilities of this API. The only thing for this particular code that comes to my mind right now is that we'd be able to hook this into our inspector - only to display the current state and stuff (not to actually draw a chart).

I'm trying to understand what are compelling reasons for users to refactor their code to this, arguably this is more boilerplate-y. This wouldn't be that much of a problem if people would have to write stuff like this in the beginning but if they must refactor less code lines to more code lines then we better give them something in return :p

Reply:
This is a contrived example; one obvious benefit over `useReducer` is that the actor ref can be shared:

```jsx
const Parent = () => {
  const actorRef = useSpawn(behavior);

  return <div>
    <Child actorRef={actorRef} />
    <OtherChild />
  </div>
}
```

And now `<OtherChild />` won't rerender when the state in `actorRef` changes.

-----

Another benefit is that behaviors aren't just for reducers. Future example:

```jsx
const behavior = fromPromise(() => fetch('something').then(...));

const Test = () => {
  const actorRef = useSpawn(behavior);
  const [state, _] = useActor(actorRef);

  // ...
}
```

Now you have the exact same interface for behaviors other than reducers!

Reply:
@Andarist Are you sold yet? 😜

Reply:
<img src="https://media.giphy.com/media/MdMasX8UlJ5aAEHgjz/giphy.gif"/>